### ISSUE TITLE
Invert transform is not general

### ISSUE THREAD

**Issue by ercius** [OpenChemistry/tomviz#492]
The invert transform is not general for all data types.

Most FEI mrc files are saved as unsigned integers which is not technically part of the MRC file specification. tomviz loads these as signed integers. My data typically is half negative and half positive.

To apply a general invert, the data set has to be converted to float in order to fit all data values within the bit range.

Proper general transform (although memory intensive):
```python
def transform_scalars(dataset):
    from tomviz import utils
    import numpy as np
    scalars = utils.get_scalars(dataset)
    if scalars is None:
        raise RuntimeError("No scalars found!")
    result = np.amax(scalars) - np.float32(scalars)
    utils.set_scalars(dataset, result)
```

**Comment by Hovden** [comment]
Should we be shifting the value here, or simply cast the scalars to float32()?

**Comment by Hovden** [comment]
Nevermind. I agree. Changing now.

**Comment by ercius** [comment]
Im not sure if we want to cast back to the original data type. This is possible using np.dtype() and np.astype()

```
def transform_scalars(dataset):
    from tomviz import utils
    import numpy as np

    scalars = utils.get_scalars(dataset)
    if scalars is None:
        raise RuntimeError("No scalars found!")
    
    result = np.amax(scalars) - scalars.astype(float)
    utils.set_scalars(dataset, result.astype(scalars.dtype))
```

**Comment by ercius** [comment]
Argh. I made a mistake and the previous code wont work. Its probably best just to leave it as float.

**Comment by Hovden** [comment]
Ok. Should be fixed in #493
